tests/test_tui_rust_lock_contract.py

```
@@ -0,0 +1,30 @@
+
+
+def test_every_published_rust_sdk_lock_records_runtime_sha2() -> None:
+    """Path consumers must stay --locked after cmux-sdk gains runtime hashing."""
+
+    locks = sorted((ROOT / "cmux-tui").rglob("Cargo.lock"))
+    sdk_packages: list[tuple[Path, dict[str, object]]] = []
+    for lock in locks:
+        document = tomllib.loads(lock.read_text(encoding="utf-8"))
+        for package in document.get("package", []):
+            if package.get("name") == "cmux-sdk" and package.get("version") == "1.0.0":
+                sdk_packages.append((lock, package))
```

Ruff (0.16.1)

[warning] Use `list.extend` to create a transformed list

(PERF401)